From 7c1a561666871a2dc1bea3525df32bbe7011ccb8 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:37:17 -0400
Subject: [ethernet] i40e: Remove a FW workaround

Message-id: <1409658042-22689-77-git-send-email-sassmann@redhat.com>
Patchwork-id: 90272
O-Subject: [RHEL7.1 PATCH 076/281] i40e: Remove a FW workaround
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Catherine Sullivan <catherine.sullivan@intel.com>

Remove the FW workaround to increment the number of msix vectors.

Signed-off-by: Catherine Sullivan <catherine.sullivan@intel.com>
Acked-by: Shannon Nelson <shannon.nelson@intel.com>
Tested-by: Kavindya Deegala <kavindya.s.deegala@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 98c95a6bfd79ddafe012f899e94af103f4932398)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_main.c b/drivers/net/ethernet/intel/i40e/i40e_main.c
index 9b46b4c..bf07d05 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_main.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_main.c
@@ -5231,9 +5231,6 @@ static int i40e_get_capabilities(struct i40e_pf *pf)
   }
  } while (err);
 
- /* increment MSI-X count because current FW skips one */
- pf->hw.func_caps.num_msix_vectors++;
-
  if (((pf->hw.aq.fw_maj_ver == 2) && (pf->hw.aq.fw_min_ver < 22)) ||
      (pf->hw.aq.fw_maj_ver < 2)) {
   pf->hw.func_caps.num_msix_vectors++;
-- 
1.7.1